napari/components/layerlist.py

```diff
@@ -69,10 +83,19 @@ def insert(self, index: int, value: Layer):
         new_layer.name = self._coerce_name(new_layer.name)
         super().insert(index, new_layer)
 
+        # required for deprecated layer.selected property.  remove after 0.4.9
+        new_layer._deprecated_layerlist = self
+
     @property
     def selected(self):
         """List of selected layers."""
-        return [layer for layer in self if layer.selected]
+        warnings.warn(
+            "'layers.selected' is deprecated and will be removed in >=v0.4.9.  "
+            "Please use 'layers.selection'",
+            category=DeprecationWarning,
+            stacklevel=2,
```

Comment:
```suggestion
            "'layer.selected' is deprecated and will be removed in or after"
            " version 0.4.9. Please use 'viewer.layers.selection'",
            category=FutureWarning,
            stacklevel=2,
```

Note that DeprecationWarnings are not user-visible by default.